src/config.rs

```diff
@@ -0,0 +1,56 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct Remote {
+    pub host: String,
+    pub user: String,
+    pub ssh_port: u16,
+    pub temp_dir: String,
+}
+
+impl Default for Remote {
+    fn default() -> Self {
+        Self {
+            host: String::new(),
+            user: String::new(),
+            ssh_port: 22,
+            temp_dir: "~/remote-builds".to_string(),
+        }
+    }
+}
+
+impl Remote {
+    pub fn user_host(&self) -> String {
+        format!("{}@{}", self.user, self.host)
+    }
+}
+
+#[derive(Debug, Default, Deserialize, Serialize)]
+pub struct Config {
+    pub remote: Remote,
+}
+
+impl Config {
+    pub fn new(project_dir: &std::path::Path) -> Result<Self, config::ConfigError> {
+        let mut conf = config::Config::default();
+
+        conf.merge(config::File::from_str(
```

Comment:
LOL! Sad this has to be so inelegant. Another option would be serializing into a `toml::Value` and implementing `config::Source` for its table variant, but that seems like more code with questionable benefits.

Reply:
I think you don't like the merge of `Config::default`. Me too.
An alternative, which should probably work is [serde-piecewise-default](https://crates.io/crates/serde_piecewise_default) but I didn't wanted to add a lot of Crates in a foreign project.

Reply:
Yeah, this is definitely better than adding yet another cdependency :smile: it's just the sillyness of serializing and then deserializing again xD but it's really not important given that way more latency will come from the network anyway.

Reply:
I did the piecewise-default by hand to regulate what should be optional.